### PR TITLE
Fix bounds on floating point range testing

### PR DESCRIPTION
**Pull request by simonbyrne**
Follow up on #48465.

Addresses CI failure noted here https://github.com/JuliaLang/julia/pull/48465#issuecomment-1419261473.

### CODE REVIEW

**Review by vtjnash — COMMENTED** [review]
Thanks!

**Comment by simonbyrne** [comment]
Can I get a review on this?

File: test/ranges.jl [hunk 1]
```diff
@@ -887,7 +887,15 @@ function range_fuzztests(::Type{T}, niter, nrange) where {T}
...
-        @test stop ≈ last(r) atol = eps((n-1)*Δ) + eps(stop) # account for intermediate rounding in computation of stop
+        # potential floating point error:
+        #   stop = strt + (n-1)*Δ
+        #      *        => eps((n-1)*Δ)/2
+        #      +        => eps(stop)/2
+        #   last(r)
+        #     rat(strt) => eps(strt)/2
+        #     rat(Δ)    => (n-1)*eps(Δ)/2
+        #     T(...)    => eps(last(r))/2
+        @test stop ≈ last(r) atol = eps((n-1)*Δ)/2 + eps(stop)/2 + eps(strt)/2 + (n-1)*eps(Δ)/2 + eps(last(r))/2
```
Comment (oscardssmith) [inline test/ranges.jl:898]: I think this tolerance can be simplified significantly.

Comment (simonbyrne) [reply test/ranges.jl:898]: If you wanted a weaker bound, you can use `eps(x) <= eps(T) * abs(x)` and so you can have:
```suggestion
        @test stop ≈ last(r) atol = (abs(strt)/2 + (n-1)*abs(Δ) + abs(stop)) * eps(T)
```
is that what you had in mind?

Comment (oscardssmith) [reply test/ranges.jl:898]: yeah, that looks a lot better.

File: test/ranges.jl [hunk 2]
```diff
@@ -887,7 +887,15 @@ function range_fuzztests(::Type{T}, niter, nrange) where {T}
...
         @test Δ == step(r)
-        @test stop ≈ last(r) atol = eps((n-1)*Δ) + eps(stop) # account for intermediate rounding in computation of stop
+        # potential floating point error:
+        #   stop = strt + (n-1)*Δ
+        #      *        => eps((n-1)*Δ)/2
+        #      +        => eps(stop)/2
+        #   last(r)
+        #     rat(strt) => eps(strt)/2
+        #     rat(Δ)    => (n-1)*eps(Δ)/2
+        #     T(...)    => eps(last(r))/2
```
Comment (simonbyrne) [inline test/ranges.jl:897]:
```suggestion
        #   stop = strt + (n-1)*Δ
        #      *          error <= eps((n-1)*Δ)/2 <= abs((n-1)*Δ)/2 * eps(T)
        #      +          error <= eps(stop)/2    <= abs(stop)/2    * eps(T)
        #   last(r)
        #     rat(strt)   error <= eps(strt)/2    <= abs(strt)/2    * eps(T)
        #     rat(Δ)      error <= (n-1)*eps(Δ)/2 <= abs((n-1)*Δ)/2 * eps(T)
        #     T(...)      error <= eps(last(r))/2 <= abs(stop)/2    * eps(T)
```

**Pull request merged**